src/Stripe.net/Infrastructure/Public/StripeClient.cs

```diff
@@ -185,6 +188,16 @@ public class StripeClient : IStripeClient
                 throw new InvalidOperationException("content is not allowed for non-POST requests.");
             }
 
+            if (requestOptions == null)
```

Comment:
I wonder if we care enough to clone the options here. The `Usage` property is not accessible by customers.

Reply:
I'm thinking about having this request conning code in every generated method (when we decide to track stripe client) and I'm not a fan.

Should we add a helper extension method? `options = options.WithUsage(...)`?

Reply:
Works for me!

Reply:
Attempted in [RawRequestOptions.WithUsage helper](https://github.com/stripe/stripe-dotnet/pull/2830/commits/9a761e4b5ba97e3f06bc469f8b6396133d9ce14b).